presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -0,0 +1,1002 @@
+            }
+            while (request.getNextToken() != null);
+
+            return Optional.of(tableNames);
+        }
+        catch (EntityNotFoundException e) {
+            // database does not exist
+            return Optional.empty();
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public Optional<List<String>> getAllViews(MetastoreContext metastoreContext, String databaseName)
+    {
+        try {
+            List<String> views = new ArrayList<>();
+            GetTablesRequest request = new GetTablesRequest().withCatalogId(catalogId).withDatabaseName(databaseName);
+
+            do {
+                GetTablesResult result = stats.getGetTables().record(() -> glueClient.getTables(request));
+                request.setNextToken(result.getNextToken());
+                result.getTableList().stream()
+                        .filter(table -> VIRTUAL_VIEW.name().equals(table.getTableType()))
+                        .forEach(table -> views.add(table.getName()));
+            }
+            while (request.getNextToken() != null);
+
+            return Optional.of(views);
+        }
+        catch (EntityNotFoundException e) {
+            // database does not exist
+            return Optional.empty();
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public void createDatabase(MetastoreContext metastoreContext, Database database)
+    {
+        if (!database.getLocation().isPresent() && defaultDir.isPresent()) {
+            String databaseLocation = new Path(defaultDir.get(), database.getDatabaseName()).toString();
+            database = Database.builder(database)
+                    .setLocation(Optional.of(databaseLocation))
+                    .build();
+        }
+
+        try {
+            DatabaseInput databaseInput = GlueInputConverter.convertDatabase(database);
+            stats.getCreateDatabase().record(() -> glueClient.createDatabase(new CreateDatabaseRequest().withCatalogId(catalogId).withDatabaseInput(databaseInput)));
+        }
+        catch (AlreadyExistsException e) {
+            throw new SchemaAlreadyExistsException(database.getDatabaseName());
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+
+        if (database.getLocation().isPresent()) {
+            createDirectory(hdfsContext, hdfsEnvironment, new Path(database.getLocation().get()));
+        }
+    }
+
+    @Override
+    public void dropDatabase(MetastoreContext metastoreContext, String databaseName)
+    {
+        try {
+            stats.getDeleteDatabase().record(() -> glueClient.deleteDatabase(new DeleteDatabaseRequest().withCatalogId(catalogId).withName(databaseName)));
+        }
+        catch (EntityNotFoundException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)